multiplex_test.go

```diff
@@ -394,6 +394,38 @@ func TestClosing(t *testing.T) {
 	}
 }
 
+func TestCloseChan(t *testing.T) {
+	a, b := net.Pipe()
+
+	mpa := NewMultiplex(a, false)
+	mpb := NewMultiplex(b, true)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	_, err := mpb.NewStream(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = mpa.Accept()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	go func() {
+		time.Sleep(time.Second * 1)
+		mpa.Close()
+	}()
+
+	select {
+	case <-ctx.Done():
+		t.Fatal("did not receive from CloseChan within timeout")
+	case <-mpb.CloseChan():
+	case <-mpa.CloseChan():
```

Comment:
Should this be a second `select` statement, to make sure that both `mpa` and `mpb` are closed?

Reply:
You are right, I should probably do a second one to make sure that both got the "message"

Reply:
Fixed in force push

Reply:
would you mind closing this PR and instead look at #89 instead? Sorry for the mix up.

Reply:
GitHub should allow you to force push, and update the PR accordingly.

Reply:
Next time then ;)
Let’s continue in #89 for now, that’s easier.